.github/workflows/pr-auto-merge.yml

```
@@ -0,0 +1,33 @@
+on:
+  pull_request:
+
+jobs:
+  e2e:
+    uses: ./.github/workflows/_e2e.yaml
+
+  merge_pr_when_e2e_success:
+    name: Merge PR when E2E success
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - e2e
+    if: ${{ always() && needs.e2e.result == 'success' && contains('["dependabot","github-actions"]', github.event.pull_request.user.login) }}
+    steps:
+      - name: Merge PR
+        run: |
+          gh pr merge --merge --auto "${{github.event.pull_request.html_url}}"
+
+  notify_slack_when_e2e_failed:
+    name: Notify Slack when E2E failed
+    runs-on: ubuntu-latest
+    needs:
+      - e2e
+    if: ${{ always() && needs.e2e.result == 'failure' && contains('["dependabot","github-actions"]', github.event.pull_request.user.login) }}
```

Sorry I removed hlts2 suggestion by mistake.
The original suggestion was:
```suggestion
    needs: [e2e]
```

---

I think we need the following suggestion to send notification only if e2e is failed.
```suggestion
    needs:
      - e2e
    if: ${{ always() && needs.e2e.result == 'failure' }}
```

---

Also what I thought was that the slack notification will be fired only when the PR creator is from CI user, to reduce the notification from any other changes from user causing e2e failed. 
Your suggestion will notify user whenever the e2e is failed, even in PR or e2e on release.

What do you think about it?